modules/internal/ChapelRange.chpl

```diff
@@ -1722,6 +1705,76 @@ operator :(r: range(?), type t: range(?)) throws where !chpl_castIsSafe(r, t) {
   return tmp;
 }
 
+/* Cast a range to a new range type.  The overload below throws when
+   the original bounds and/or stride do not fit in the new type or 'strides'.
+   TODO: should we allow 't' to be generic?
+ */
+@chpldoc.nodoc
+operator :(r: range(?), type t: range(?)) where chpl_castIsSafe(r, t) {
+  // note: the implementation is duplicated with the throwing overload
```

Comment:
If this is the same as tryCast, why not call `tryCast`?

Reply:
It was `operator :` prior to this PR that behaved the same as the new `tryCast`. With this PR, `operator :` and `tryCast` behave differently.